### PR TITLE
CR-9845 cluster remove

### PR DESCRIPTION
**Pull request by rotem-codefresh**

### CODE REVIEW

File: cmd/commands/cluster.go [hunk 1]
```diff
@@ -187,6 +193,58 @@ func createAddClusterKustomization(ingressUrl, contextName, server, csdpToken, v
...
+
+			ctx := cmd.Context()
+
+			opts.runtimeName, err = ensureRuntimeName(ctx, args)
+			if err != nil {
+				return err
+			}
+
+			if opts.server == "" {
+				return fmt.Errorf("must supply value for the cluster's server url with --server-url flag")
```
Comment (ghost) [inline cmd/commands/cluster.go]: you can just mark it as required when you declare the flag

File: cmd/commands/cluster.go [hunk 2]
```diff
@@ -187,6 +193,58 @@ func createAddClusterKustomization(ingressUrl, contextName, server, csdpToken, v
...
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.server, "server-url", "", "The cluster's server url")
+
+	return cmd
+}
+
+func runClusterRemove(ctx context.Context, opts *ClusterRemoveOptions) error {
+	appProxy, err := cfConfig.NewClient().AppProxy(ctx, *&opts.runtimeName, store.Get().InsecureIngressHost)
```
Comment (ghost) [inline cmd/commands/cluster.go]: why does it have `*&opts.runtimeName` ?

File: cmd/commands/cluster.go [hunk 3]
```diff
@@ -187,6 +193,58 @@ func createAddClusterKustomization(ingressUrl, contextName, server, csdpToken, v
 	return k
 }
 
+func NewClusterRemoveCommand() *cobra.Command {
+	var (
+		opts ClusterRemoveOptions
+	)
+
+	cmd := &cobra.Command{
+		Use:     "remove RUNTIME_NAME",
```
Comment (roi-codefresh) [inline cmd/commands/cluster.go:202]:
```suggestion
		Use:     "remove CLUSTER_NAME",
```

File: cmd/commands/cluster.go [hunk 4]
```diff
@@ -187,6 +193,58 @@ func createAddClusterKustomization(ingressUrl, contextName, server, csdpToken, v
...
+
+func runClusterRemove(ctx context.Context, opts *ClusterRemoveOptions) error {
+	appProxy, err := cfConfig.NewClient().AppProxy(ctx, *&opts.runtimeName, store.Get().InsecureIngressHost)
+	if err != nil {
+		return err
+	}
+
+	err = appProxy.AppProxyClusters().RemoveCluster(ctx, opts.server, opts.runtimeName)
+	if err != nil {
+		return fmt.Errorf("failed to get app-proxy client, cluster was not removed")
```
Comment (ghost) [inline cmd/commands/cluster.go]: the error message isn't very clear. `failed to get app-proxy client` ?

File: cmd/commands/runtime.go [hunk 5]
```diff
@@ -605,11 +605,11 @@ func RunRuntimeInstall(ctx context.Context, opts *RuntimeInstallOptions) error {
 
 	componentNames := getComponents(rt, opts)
 
-	shouldRollback := opts.DisableRollback
+	disableRollback := opts.DisableRollback
```
Comment (ghost) [inline cmd/commands/runtime.go:608]: why reverse the flag name and meaning?

File: cmd/commands/cluster.go (hunk 3 above)
Comment (rotem-codefresh) [reply cmd/commands/cluster.go:202]: it's suppose to be runtime name

**Review by noam-codefresh — APPROVED** [review]

**Pull request merged**